paddle/phi/infermeta/spmd_rules/matmul.cc

```diff
@@ -221,24 +221,16 @@ MatmulSPMDRule::InferForward(const std::vector<DistTensorSpec>& input_specs,
   return {{x_dist_attr_dst, y_dist_attr_dst}, {output_dist_attr_dst}};
 }
 
-std::pair<std::vector<TensorDistAttr>, std::vector<TensorDistAttr>>
-MatmulSPMDRule::InferBackward(const std::vector<DistTensorSpec>& input_specs,
-                              const std::vector<DistTensorSpec>& output_specs,
-                              const paddle::framework::AttributeMap& attrs) {
-  // extra & verify input
-  auto output_specs_size = output_specs.size();
-  PADDLE_ENFORCE_EQ(
-      output_specs_size,
-      1,
-      phi::errors::InvalidArgument(
-          "The size of OutputSpec of matmul should be 1, but got [%d].",
-          output_specs_size));
-
-  auto out_shape = output_specs[0].shape();
+SpmdInfo MatmulSpmdInferBackward(const DistMetaTensor& x,
+                                 const DistMetaTensor& y,
+                                 const DistMetaTensor& out,
```

Comment:
for variadic op like split and concat， should use vector for the variadic slot ？

Phi api for concat：
PADDLE_API Tensor concat(const std::vector<Tensor>& x, const Scalar& axis)

spmd for  concat:
SpmdInfo ConcatSpmdInferBackward(const std::vector<DistMetaTensor>& x, const DistMetaTensor& out, const Scalar& axis)

AND for  ReplicatedSpmd Rule which is a bottom line rule for all Ops that have not specific rule:
SpmdInfo ReplicatedSpmdInferBackward(const std::vector<DistMetaTensor>& x, const std::vector<DistMetaTensor>&out)

Reply:
same as above

for ReplicatedSpmd Rule, we can use the general format 
```
SpmdInfo ReplicatedSpmdInferBackward(const std::vector<const DistMetaTensor*>& x, const std::vector<const DistMetaTensor*>& out, const std::vector<phi::Attribtue>& attrs)
```
we also can unify its format into `SpmdInfo (*)(const InferSpmdContext&)`